li-apache-kafka-clients/src/main/java/com/linkedin/kafka/clients/largemessage/ConsumerRecordsProcessor.java

```diff
@@ -87,6 +87,15 @@ boolean isCommittedOffsetDefined() {
   private final DeserializeStrategy<V> _deserializeStrategy;
 
   private long recordsSkipped = 0;
+  /**
+   * Gives an idea about how far behind the safe offset of a partition is from the watermark. Hence, the
+   * offset-watermark span or a partition is computed as (watermark minus safeoffset).
+   * This metric is a sum of
+   * {@link com.linkedin.kafka.clients.largemessage.DeliveredMessageOffsetTracker#offsetWatermarkSpan} for all
+   * assigned partitions that have currently undelivered large messages in the consumer. It is computed on query of
+   * the metric each time and hence, stores the last known value for this metric.
+   */
+  private long _consumerOffsetWatermarkSpan = -1;
```

Comment:
nit - wouldnt 0 be a safer default?

Reply:
Probably.  I think this local variable is not required as the metric is computed in each call to `getConsumerOffsetWatermarkSpan`